engine/src/main/java/org/terasology/rendering/dag/RenderTaskListGenerator.java

```diff
@@ -227,4 +228,25 @@ public void refresh() {
         generateFrom(nodeList);
     }
 
+    /**
+     * Refreshes the task list if a refresh has been requested.
+     */
+    public void refreshIfNeeded() {
+        if (refreshRequested) {
+            refresh();
+
+            // Clear refresh requests to prevent refreshing unless needed
+            refreshRequested = false;
+        }
+    }
```

Comment:
Can be worked around by using `CoreRegistry.get()` or adding a `WorldRenderer` to `AbstractNode` but the first one is a hack and the second one is a bit too liberal, but is the best option. Going with the latter.